tests/test_price_parsing.py

```diff
@@ -1974,7 +1974,7 @@ def __eq__(self, other):
 )
 def test_parsing(example: Example):
     parsed = Price.fromstring(example.price_raw, example.currency_raw)
-    assert parsed == example
+    assert parsed == example, f"Failed scenario: price={example.price_raw}, currency_hint={example.currency_raw}"
```

Comment:
I wonder if it would be better to use pytest ids for this - see https://docs.pytest.org/en/latest/example/parametrize.html#different-options-for-test-ids. We can use Example as an id; this would also allow to have more readable test names for tests which are passing.

Reply:
The representation of `Example` is useless. As it inherits from `Price`, its string representation doesn't contains the input arguments to `fromstring` method. IMO this is the most important information that is missing when our tests fails.

This is what we get from `Example` repr:
```
ipdb> repr(value)                                                                                                                                                           
"Example(amount=Decimal('12.99'), currency='US$')"
```
As you can see, it shows nothing about the the input `US$:12.99`.

But these are the information that we really need when reading the tests fails:
```
ipdb> value.currency_raw, value.price_raw                                                                                                                                   
(None, 'US$:12.99')
```

We could create an id based on the raw values. It is better than the actual error, but as we have limitations over the characters that can be user as an id (considering that the inputs has spaces and special characters), it won't contain more detailed information about the inputs.

Another option I can see is to define `__repr__` of `Example` class, so it returns all values provided.

However, I believe that the only values that really matter when we see an error are the ones passed to `.fromstring` method.

Reply:
ok, fair enough,  thanks for the explanation @rennerocha!